sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
```

Comment:
I don't like assert here, that would mean that you can crash the system if a driver is using a scale which just doesn't have a simple prefix. 10 m^2 could just as well be represented as value 1 and scale 1 or value 10 and scale 0, the system should not crash when you try to print it regardless.